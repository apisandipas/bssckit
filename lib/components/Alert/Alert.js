"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.AlertDismissIcon = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _base = require("../Utilities/base");

var _themeFunctions = require("../../utils/themeFunctions");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _templateObject11() {
  var data = _taggedTemplateLiteral(["\n  position: relative;\n  padding: ", ";\n  border: ", ";\n  & > a {\n    font-weight: ", ";\n  };\n  & > h1,\n  & > h2,\n  & > h3,\n  & > h4,\n  & > h5,\n  & > h6 {\n    color: inherit;\n  };\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteral(["\n  font-size: 2rem;\n  cursor: pointer;\n  line-height: 1rem;\n  position: absolute;\n  top: 1rem;\n  right: 1rem;\n  color: inherit;\n  opacity: 0.5;\n\n  &:hover,\n  &:focus {\n    opacity: 0.75;\n  }\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["\n    border-radius: ", ";\n  "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["\n      border-radius: ", ";\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n      border-radius: ", ";\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n    & > a,\n    & > a:hover {\n      color: ", ";\n    }\n  "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    & > hr {\n      border-top-color: ", ";\n    }\n  "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n    border-color: ", ";\n  "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    background-color: ", ";\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    color: ", ";\n  "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    padding-right: ", ";\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var alertDismissible = function alertDismissible(props) {
  return props.dismissible && (0, _styledComponents.css)(_templateObject(), (0, _themeFunctions.getPadding)(props, 'alert', 'dismissibleRight'));
};

var color = function color(props) {
  return (0, _styledComponents.css)(_templateObject2(), (0, _themeFunctions.getColor)(props, 'alert', 'text'));
};

var backgroundColor = function backgroundColor(props) {
  return (0, _styledComponents.css)(_templateObject3(), (0, _themeFunctions.getColor)(props, 'alert', 'background'));
};

var borderColor = function borderColor(props) {
  return (0, _styledComponents.css)(_templateObject4(), (0, _themeFunctions.getColor)(props, 'alert', 'border'));
};

var borderTopColor = function borderTopColor(props) {
  return (0, _styledComponents.css)(_templateObject5(), (0, _themeFunctions.getColor)(props, 'alert', 'borderTop'));
};

var linkHoverColor = function linkHoverColor(props) {
  return (0, _styledComponents.css)(_templateObject6(), (0, _themeFunctions.getColor)(props, 'alert', 'hover'));
};

var borderRadius = function borderRadius(props) {
  if (props.noRadius) {
    return (0, _styledComponents.css)(_templateObject7(), (0, _themeFunctions.getBorderRadius)(props, 'alert', 'noRadius'));
  } else if (props.pill) {
    return (0, _styledComponents.css)(_templateObject8(), (0, _themeFunctions.getBorderRadius)(props, 'alert', 'pill'));
  }

  return (0, _styledComponents.css)(_templateObject9(), (0, _themeFunctions.getBorderRadius)(props, 'alert', 'default'));
};

var AlertDismissIcon = _styledComponents["default"].span.attrs({
  ariaHidden: 'true'
})(_templateObject10());

exports.AlertDismissIcon = AlertDismissIcon;
var Alert = (0, _styledComponents["default"])(_base.Div).attrs({
  role: 'alert'
})(_templateObject11(), function (props) {
  return (0, _themeFunctions.getPadding)(props, 'alert', 'default');
}, function (props) {
  return (0, _themeFunctions.getBorder)(props, 'alert', 'default');
}, function (props) {
  return (0, _themeFunctions.getFontWeight)(props, 'alert', 'default');
}, function (props) {
  return borderRadius(props);
}, function (props) {
  return alertDismissible(props);
}, function (props) {
  return color(props);
}, function (props) {
  return backgroundColor(props);
}, function (props) {
  return borderColor(props);
}, function (props) {
  return borderTopColor(props);
}, function (props) {
  return linkHoverColor(props);
});
var _default = Alert;
exports["default"] = _default;