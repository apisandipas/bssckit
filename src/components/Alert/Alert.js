import styled, { css } from 'styled-components';
import { theme, ifProp } from 'styled-tools';
import { Div } from '../base';
import defaultTheme from '../../utils/defaultTheme';
import { themeContexts } from '../../utils/consts';

export const AlertDismissIcon = styled.span.attrs({
  ariaHidden: 'true',
})`
  font-size: 2rem;
  cursor: pointer;
  line-height: 1rem;
  position: absolute;
  top: 1rem;
  right: 1rem;
  color: inherit;
  opacity: 0.5;

  &:hover,
  &:focus {
    opacity: 0.75;
  }
`;

const Alert = styled(Div).attrs({
  role: 'alert',
})`
  position: relative;
  padding: ${theme('alert.padding.default')};
  margin-bottom: ${theme('alert.margin.bottom')};
  border: ${theme('alert.border.default')};
  border-radius: ${theme('alert.borderRadius.default')};
  color: ${theme(`alert.colors.default.color`)};
  background-color: ${theme(`alert.colors.default.background`)};
  border-color: ${theme(`alert.colors.default.border`)};

  & > a,
  & > a:hover {
    font-weight: ${theme('alert.fontWeight.default')};
  }

  & > h1,
  & > h2,
  & > h3,
  & > h4,
  & > h5,
  & > h6 {
    color: inherit;
  }

  ${themeContexts.map(context => {
    return ifProp(
      context,
      css`
        color: ${theme(`alert.colors.text`)};
        background-color: ${theme(`alert.colors.${context}.background`)};
        border-color: ${theme(`alert.colors.${context}.border`)};

        & > a,
        & > a:hover {
          color: ${theme(`alert.colors.${context}.text`)};
        }

        & > h1,
        & > h2,
        & > h3,
        & > h4,
        & > h5,
        & > h6 {
          color: ${theme(`alert.colors.${context}.text`)};
        }

        & > hr {
          border-top-color: ${theme(`alert.colors.${context}.text`)};
        }
      `
    );
  })}

  ${ifProp(
    'noRadius',
    css`
      border-radius: 0;
    `
  )}

  ${ifProp(
    'pill',
    css`
      border-radius: ${theme('alert.borderRadius.pill')};
    `
  )}

  ${ifProp(
    'dismissible',
    css`
      padding-right: ${theme('alert.padding.dismissibleRight')};
    `
  )};
`;

Alert.defaultProps = {
  theme: defaultTheme,
};

export default Alert;
