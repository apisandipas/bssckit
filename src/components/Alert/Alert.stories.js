import React, { useState } from 'react';
import { Alert, AlertDismissIcon, DismissableAlert } from './';
import { Story } from '../../utils/functions';
import BSThemeProvider from '../Utilities/BSThemeProvider';

export default {
  title: 'Alert',
};

export const simple = () => <Alert primary>Hello World</Alert>;

export const noRadius = () => (
  <Alert secondary noRadius>
    Hello World
  </Alert>
);

export const pill = () => (
  <Alert info pill>
    Hello World
  </Alert>
);

export const contextual = () => (
  <>
    <Alert primary mb4>
      Primary alert
    </Alert>
    <Alert secondary mb4>
      Secondary alert
    </Alert>
    <Alert success mb4>
      Success alert
    </Alert>
    <Alert danger mb4>
      Danger alert
    </Alert>
    <Alert warning mb4>
      Warning alert
    </Alert>
    <Alert info mb4>
      Info alert
    </Alert>
    <Alert light mb4>
      Light alert
    </Alert>
    <Alert dark>Dark alert</Alert>
  </>
);

export const withALink = () => (
  <>
    <Alert primary mb4>
      This is a primary alert with <a href='#'>an example link</a>. Give it a click if you like.
    </Alert>
    <Alert danger>
      This is a primary alert with <a href='#'>an example link</a>. Give it a click if you like.
    </Alert>
  </>
);

export const staticDismissible = () => (
  <>
    <Alert primary dismissible>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Iste tempore ea distinctio iure quas quibusdam
      dolorem inventore dolorum explicabo a aspernatur quisquam tempora, sapiente sed repudiandae, iusto
      architecto ipsum blanditiis.
      <AlertDismissIcon>&times;</AlertDismissIcon>
    </Alert>
  </>
);

export const dismissible = Story(() => {
  const [hidden, setHidden] = useState(false);
  return (
    <DismissableAlert danger toggle={() => setHidden(!hidden)} isOpen={!hidden}>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Iste tempore ea distinctio iure quas quibusdam
      dolorem inventore dolorum explicabo a aspernatur quisquam tempora, sapiente sed repudiandae, iusto
      architecto ipsum blanditiis.
    </DismissableAlert>
  );
});

const theme = {
  alert: {
    colors: {
      default: {
        text: 'white',
        background: 'cornflowerblue',
      },
    },
  },
};

export const withTheme = () => (
  <BSThemeProvider theme={theme}>
    <Alert>Always blue! Always blue!</Alert>
  </BSThemeProvider>
);

const theme2 = {
  palette: {
    primaryLight: '#bada55',
    dangerLight: '#db84be',
  },
};

export const withPaletteOverride = () => (
  <BSThemeProvider theme={theme2}>
    <Alert primary mb4>
      Testing 123!
    </Alert>
    <Alert danger>Testing 123!</Alert>
  </BSThemeProvider>
);
