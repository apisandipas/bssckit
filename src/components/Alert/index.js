import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { theme, ifProp } from 'styled-tools';
import { Div } from '../base';
export { default as DismissableAlert } from './DismissableAlert';
import defaultTheme from '../../utils/defaultTheme';
import { themeContexts } from '../../utils/consts';

const { alert } = defaultTheme;

export const AlertDismissIcon = styled.span.attrs({
  ariaHidden: 'true',
})`
  font-size: 2rem;
  cursor: pointer;
  line-height: 1rem;
  position: absolute;
  top: 1rem;
  right: 1rem;
  color: inherit;
  opacity: 0.5;

  &:hover,
  &:focus {
    opacity: 0.75;
  }
`;

const Alert = styled(Div).attrs({
  role: 'alert',
})`
  position: relative;
  padding: ${theme('alert.padding.default', alert.padding.default)};
  margin-bottom: ${theme('alert.margin.bottom', alert.margin.bottom)};
  border: ${theme('alert.border', alert.border.default)};
  border-radius: ${theme('alert.borderRadius.default', alert.borderRadius.default)};

  & > a,
  & > a:hover {
    font-weight: ${theme('alert.fontWeight.default', alert.fontWeight.default)};
  }

  & > h1,
  & > h2,
  & > h3,
  & > h4,
  & > h5,
  & > h6 {
    color: inherit;
  }

  ${themeContexts.map(context => {
    const { text, background, border } = defaultTheme.alert.colors[context];

    return ifProp(
      context,
      css`
        color: ${theme(`colorScheme.${context}Darkest`, text)};
        background-color: ${theme(`colorScheme.${context}Light`, background)};
        border-color: ${theme(`colorScheme.${context}Lighter`, border)};

        & > a,
        & > a:hover {
          color: ${theme(`colorScheme.${context}Darkest`, text)};
        }

        & > h1,
        & > h2,
        & > h3,
        & > h4,
        & > h5,
        & > h6 {
          color: ${theme(`colorScheme.${context}Darkest`, text)};
        }

        & > hr {
          border-top-color: ${theme(`colorScheme.${context}Darkest`, text)};
        }
      `
    );
  })}

  ${ifProp(
    'noRadius',
    css`
      border-radius: 0;
    `
  )}

  ${ifProp(
    'pill',
    css`
      border-radius: 10rem;
    `
  )}

  ${ifProp(
    'dismissible',
    css`
      padding-right: ${theme('alert.padding.dismissibleRight', alert.padding.dismissibleRight)};
    `
  )};
`;

export default Alert;
