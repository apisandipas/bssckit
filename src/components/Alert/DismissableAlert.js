import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Alert, { AlertDismissIcon } from './Alert';
import { tagPropType } from '../../utils/consts';
import Fade from '../../utils/Fade';

const DismissButton = styled.button`
  padding: 0;
  background-color: transparent;
  border: 0;
  appearance: none;
`;

const DismissableAlert = props => {
  const {
    className,
    closeClassName,
    closeAriaLabel,
    tag: Tag,
    isOpen,
    toggle,
    children,
    transition,
    fade,
    innerRef,
    ...attributes
  } = props;

  const alertTransition = {
    ...Fade.defaultProps,
    ...transition,
    baseClass: fade ? transition.baseClass : '',
    timeout: fade ? transition.timeout : 0,
  };

  return (
    <Fade {...attributes} {...alertTransition} tag={Alert} in={isOpen} role='alert' innerRef={innerRef}>
      {children}
      {toggle ? (
        <DismissButton type='button' aria-label={closeAriaLabel} onClick={toggle}>
          <AlertDismissIcon aria-hidden='true' aria-label={closeAriaLabel} onClick={toggle}>
            &times;
          </AlertDismissIcon>
        </DismissButton>
      ) : null}
    </Fade>
  );
};

DismissableAlert.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  closeClassName: PropTypes.string,
  closeAriaLabel: PropTypes.string,
  color: PropTypes.string,
  fade: PropTypes.bool,
  isOpen: PropTypes.bool,
  toggle: PropTypes.func,
  tag: tagPropType,
  transition: PropTypes.shape(Fade.propTypes),
  innerRef: PropTypes.oneOfType([PropTypes.object, PropTypes.string, PropTypes.func]),
};

DismissableAlert.defaultProps = {
  dismissible: true,
  isOpen: true,
  tag: Alert,
  closeAriaLabel: 'Close',
  fade: true,
  transition: {
    ...Fade.defaultProps,
    unmountOnExit: true,
  },
};

export default DismissableAlert;
