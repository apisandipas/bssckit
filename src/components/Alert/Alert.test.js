import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Alert from './index';

test('Styles match plain', () => {
  const tree = renderer.create(<Alert>Hello World</Alert>).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('position', 'relative');
  expect(tree).toHaveStyleRule('padding', '0.75rem 1.25rem');
  expect(tree).toHaveStyleRule('border', '1px solid transparent');
  expect(tree).toHaveStyleRule('border-radius', '0.25rem');
  expect(tree).toHaveStyleRule('font-weight', '700', {
    modifier: '& > a',
  });
  expect(tree).toHaveStyleRule('color', 'inherit', {
    modifier: '& > h1',
  });
  expect(tree).toHaveStyleRule('color', 'inherit', {
    modifier: '& > h2',
  });
  expect(tree).toHaveStyleRule('color', 'inherit', {
    modifier: '& > h3',
  });
  expect(tree).toHaveStyleRule('color', 'inherit', {
    modifier: '& > h4',
  });
  expect(tree).toHaveStyleRule('color', 'inherit', {
    modifier: '& > h5',
  });
  expect(tree).toHaveStyleRule('color', 'inherit', {
    modifier: '& > h6',
  });
});

test('Styles match primary', () => {
  const tree = renderer.create(<Alert primary />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Styles match secondary', () => {
  const tree = renderer.create(<Alert secondary />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Styles match success', () => {
  const tree = renderer.create(<Alert success />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Styles match danger', () => {
  const tree = renderer.create(<Alert danger />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Styles match warning', () => {
  const tree = renderer.create(<Alert warning />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Styles match info', () => {
  const tree = renderer.create(<Alert info />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Styles match light', () => {
  const tree = renderer.create(<Alert light />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Styles match dark', () => {
  const tree = renderer.create(<Alert dark />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Styles match noRadius', () => {
  const tree = renderer.create(<Alert noRadius />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('border-radius', '0');
});

test('Styles match pill', () => {
  const tree = renderer.create(<Alert pill />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('border-radius', '10rem');
});
