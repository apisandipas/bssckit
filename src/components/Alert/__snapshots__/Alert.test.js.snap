// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styles matches danger 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: rgb(114,28,36);
  background-color: rgb(248,215,218);
  border-color: rgb(245,180,187);
}

.c0 > a,
.c0 > a:hover {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > a,
.c0 > a:hover {
  color: rgb(114,28,36);
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: rgb(114,28,36);
}

.c0 > hr {
  border-top-color: rgb(114,28,36);
}

<div
  className="sc-AxjAm c0"
  role="alert"
/>
`;

exports[`Styles matches dark 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: gb(23,26,29);
  background-color: rgb(214,216,217);
  border-color: rgb(171,174,177);
}

.c0 > a,
.c0 > a:hover {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > a,
.c0 > a:hover {
  color: gb(23,26,29);
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: gb(23,26,29);
}

.c0 > hr {
  border-top-color: gb(23,26,29);
}

<div
  className="sc-AxjAm c0"
  role="alert"
/>
`;

exports[`Styles matches info 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: rgb(16,112,127);
  background-color: rgb(209,236,241);
  border-color: rgb(163,214,222);
}

.c0 > a,
.c0 > a:hover {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > a,
.c0 > a:hover {
  color: rgb(16,112,127);
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: rgb(16,112,127);
}

.c0 > hr {
  border-top-color: rgb(16,112,127);
}

<div
  className="sc-AxjAm c0"
  role="alert"
/>
`;

exports[`Styles matches light 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: rgb(52,58,64);
  background-color: rgb(254,254,254);
  border-color: rgb(204,206,207);
}

.c0 > a,
.c0 > a:hover {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > a,
.c0 > a:hover {
  color: rgb(52,58,64);
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: rgb(52,58,64);
}

.c0 > hr {
  border-top-color: rgb(52,58,64);
}

<div
  className="sc-AxjAm c0"
  role="alert"
/>
`;

exports[`Styles matches noRadius 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  border-radius: 0;
}

.c0 > a,
.c0 > a:hover {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

<div
  className="sc-AxjAm c0"
  role="alert"
/>
`;

exports[`Styles matches pill 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  border-radius: 10rem;
}

.c0 > a,
.c0 > a:hover {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

<div
  className="sc-AxjAm c0"
  role="alert"
/>
`;

exports[`Styles matches plain 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
}

.c0 > a,
.c0 > a:hover {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

<div
  className="sc-AxjAm c0"
  role="alert"
>
  Hello World
</div>
`;

exports[`Styles matches primary 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: rgb(0,64,133);
  background-color: rgb(204,229,255);
  border-color: rgb(156,201,251);
}

.c0 > a,
.c0 > a:hover {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > a,
.c0 > a:hover {
  color: rgb(0,64,133);
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: rgb(0,64,133);
}

.c0 > hr {
  border-top-color: rgb(0,64,133);
}

<div
  className="sc-AxjAm c0"
  role="alert"
/>
`;

exports[`Styles matches secondary 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: rgb(56,61,65);
  background-color: rgb(226,227,229);
  border-color: rgb(180,181,183);
}

.c0 > a,
.c0 > a:hover {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > a,
.c0 > a:hover {
  color: rgb(56,61,65);
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: rgb(56,61,65);
}

.c0 > hr {
  border-top-color: rgb(56,61,65);
}

<div
  className="sc-AxjAm c0"
  role="alert"
/>
`;

exports[`Styles matches success 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: rgb(21,87,36);
  background-color: rgb(212,237,218);
  border-color: rgb(114,216,137);
}

.c0 > a,
.c0 > a:hover {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > a,
.c0 > a:hover {
  color: rgb(21,87,36);
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: rgb(21,87,36);
}

.c0 > hr {
  border-top-color: rgb(21,87,36);
}

<div
  className="sc-AxjAm c0"
  role="alert"
/>
`;

exports[`Styles matches warning 1`] = `
.c0 {
  position: relative;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: rgb(133,100,4);
  background-color: rgb(255,243,205);
  border-color: rgb(243,218,141);
}

.c0 > a,
.c0 > a:hover {
  font-weight: 700;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: inherit;
}

.c0 > a,
.c0 > a:hover {
  color: rgb(133,100,4);
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  color: rgb(133,100,4);
}

.c0 > hr {
  border-top-color: rgb(133,100,4);
}

<div
  className="sc-AxjAm c0"
  role="alert"
/>
`;
